contracts/JobFactory.sol

```diff
@@ -197,6 +207,7 @@ contract JobFactory {
         uint64 _trainingErrorRate
     ) public {
         Job memory job = jobs[_jobPoster][_id];
+        require(job.workerNode != address(0), 'Worker Node has not yet been selected');
```

Comment:
On line 141 we're setting the auction winner / worker node to the data scientist's wallet address, but it seems like we're still checking against a zero-ed out wallet address; which is from the previous implementation (previous version of this file's line 133 IIRC)

Reply:
Line 141 is setting the end-user as the highest bidder within the Auction mapping. However, this check is ensuring that the worker-node is found within the Jobs mapping. State of workloads is tracked together with Auctions[] and Jobs[].